crates/gitbutler-core/src/git/commit_ext.rs

```diff
@@ -0,0 +1,46 @@
+// use anyhow::Result;
+use super::Result;
+use bstr::BStr;
+use git2::Commit;
+
+/// Extension trait for `git2::Commit`.
+///
+/// For now, it collects useful methods from `gitbutler-core::git::Commit`
+pub trait CommitExt {
+    /// Obtain the commit-message as bytes, but without assuming any encoding.
+    fn message_bstr(&self) -> &BStr;
+    fn change_id(&self) -> Option<String>;
+    fn parents_gb(&self) -> Result<Vec<Commit<'_>>>;
+    fn is_signed(&self) -> bool;
+    fn tree_gb(&self) -> Result<git2::Tree<'_>>;
+}
+
+impl<'repo> CommitExt for git2::Commit<'repo> {
+    fn message_bstr(&self) -> &BStr {
+        self.message_bytes().as_ref()
+    }
+    fn change_id(&self) -> Option<String> {
+        let cid = self.header_field_bytes("change-id").ok()?;
+        if cid.is_empty() {
+            None
+        } else {
+            // convert the Buf to a string
+            let ch_id = std::str::from_utf8(&cid).ok()?.to_owned();
+            Some(ch_id)
+        }
+    }
+    fn parents_gb(&self) -> Result<Vec<Commit<'repo>>> {
```

Comment:
fyi @Byron i struggled with naming this... It cant be parents because it clashes with the `git2::Commit.parents` method...
Is there some convention? But also, what is the point of this method.... perhaps we just use the one from git2 that returns an iteratior

Reply:
Yes, definitely, there is no need for this one. An iterator is better here. People who want a Vec do `commit.parents().collect()`.